rslib/src/notetype/checks.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::{borrow::Cow, fmt::Write, ops::Deref};
+
+use anki_i18n::without_unicode_isolation;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use super::CardTemplate;
+use crate::prelude::*;
+
+#[derive(Debug, PartialEq)]
+struct Template<'a> {
+    notetype: &'a str,
+    card_type: &'a str,
+    front: bool,
+}
+
+lazy_static! {
+    static ref MEDIA_FIELD_REFERENCE: Regex = Regex::new(
+        r#"(?xsi)
+            # an image, audio, or object tag
+            <\b(?:img|audio|object)\b[^>]+\b(?:src|data)\b=
+            (?:
+                    # 1: double-quoted
+                    "
+                    \{\{.+?\}\}
+                    "
+                    [^>]*>                    
+                |
+                    # 2: single-quoted
+                    '
+                    \{\{.+?\}\}
+                    '
+                    [^>]*>
+                |
+                    # 3: unquoted
+                    \{\{.+?\}\}
+                    (?:
+                        # then either a space and the rest
+                        \x20[^>]*>
+                        |
+                        # or the tag immediately ends
+                        >
+                    )
+            )
+        |
+            # an Anki sound tag
+            \[sound:\{\{.+?\}\}\]
+        |
+            # standard latex
+            \[latex\]\{\{.+?\}\}\[/latex\]
+        |
+            # inline math
+            \[\$\]\{\{.+?\}\}\[/\$\]
+        |
+            # math environment
+            \[\$\$\]\{\{.+?\}\}\[/\$\$\]
+        "#
+    )
+    .unwrap();
+}
```

Comment:
Two related issues:

- It would be nice if we were able to reuse HTML_MEDIA_TAGS, LATEX and SOUND_TAG instead of duplicating them here. 
- The field reference may be only part of the reference, eg:

[sound:{{Front}}.mp3]

Maybe we could loop through the existing regexs, extracting the referenced filenames, and then checking if they contain a field reference?

Reply:
I thought too much in terms of speed. I see how this makes more sense.